src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -133,13 +133,32 @@ private BoundExpression VisitCollectionInitializerCollectionExpression(BoundColl
         {
             Debug.Assert(!_inExpressionLambda);
 
-            var rewrittenReceiver = VisitExpression(node.CollectionCreation);
+            var elements = node.Elements;
+            BoundExpression? rewrittenReceiver;
+
+            // If the collection type is List<T> and there are no spread elements, use .ctor(int capacity).
+            // We should expand this optimization to allow spread elements with known length as well.
+            if (ShouldUseKnownLength(node, out int numberIncludingLastSpread) &&
+                numberIncludingLastSpread == 0 &&
+                elements.Length > 0 &&
+                collectionType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_List_T)))
+            {
+                // List<ElementType> list = new(N + s1.Length + ...);
+                var constructor = ((MethodSymbol)_factory.WellKnownMember(WellKnownMember.System_Collections_Generic_List_T__ctorInt32)).AsMember((NamedTypeSymbol)collectionType);
```

Comment:
do you need to look to see if this constructor exists?  and only use it if so?  or is this ok to just do as is?

Reply:
From the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?view=net-7.0#system-collections-generic-list-1-ctor(system-int32)), this constructor is available on .NET Framework 2.0 and later.

Reply:
We're testing the missing constructor case in `KnownLength_List_MissingConstructor()`.